### PR TITLE
chore: add documentation for overall flow in README (fixes #636)

### LINKED ISSUE #636: Improving the introduction README file with very simple TLDR
_TL;DR: the readme file is harsh for newcomers, it's lacking an oversimplified overview of WTF is doing sentence-collector and in what order._

Lowering the barrier for understanding the project shall help people to come. I hope. ...And even if not, ME, I need this stuff to be written down, to be clarified and to be understandable for a dummy like me.

After (trying to) work on the PR https://github.com/common-voice/sentence-collector/pull/635 (full disclosure : I started it 😸), I am now convinced that something is missing in the introduction of the sentence collector [[README.md](https://github.com/common-voice/sentence-collector#readme)](https://github.com/common-voice/sentence-collector#readme) file. 

Let me explain in two words. 

GLOBALLY (as discussed in an other FR discussion and/or issue)
we have 
`Sentence-collector > recording & review > CorporaCreator
`

LOCALLY, I was thinking that thes steps were 
`Import start > Cleanup > Validation > Add it Common Voice database ready for recording & review. 
`
And it seems that the steps are 
`Import start > Validation > Cleanup  > Add it Common Voice database ready for recording & review. . 
`
There is a discussion [here (sentence-collector-cleanup-before-export-vs-cleanup-on-upload/105411)](https://discourse.mozilla.org/t/sentence-collector-cleanup-before-export-vs-cleanup-on-upload/105411) that seems to discuss this point... In short, I'm lost. 

### **So, MY PROPOSAL IS :** 
to add a little paragraph right before Get Involved saying : 

# Common Voice Sentence Collector
The [Sentence Collector](https://commonvoice.mozilla.org/sentence-collector/) is part of the [Common Voice](https://commonvoice.mozilla.org/) project. Its purpose is to provide a tool for contributors to upload public domain sentences, which then can get reviewed and are exported to the Common Voice database. Once imported they will show up for contributors on Common Voice to read out aloud.
## Quick overview
Sentence-collector is the "entry point" for data to be [recorded and reviewed](https://github.com/common-voice/common-voice). Then, once ready, it will go to [CorporaCreator](https://github.com/common-voice/CorporaCreator).

Under the hood, 
Sentence-collector work like this (oversimplified version) : 

- You take a buch of sentence you want to add the the database (as explained [here](https://commonvoice.mozilla.org/fr/about?tab=how-add-sentences)), 
- then it's [cleaned up](https://github.com/common-voice/sentence-collector/tree/main/server/lib/cleanup), 
- after, it's [validated](https://github.com/common-voice/sentence-collector/tree/main/server/lib/validation), 
- and finally sent to the text database for recording & review. 

We can always go deeper in documentation in [common-voice docs](https://github.com/common-voice/common-voice/tree/main/docs).
## Get involved
(...)

### PR DESCRIPTION
**Pull request by MichaelKohler**

### CODE REVIEW

**Comment by MichaelKohler** [comment]
@CapitainFlam could you have a look at the changes here if they make this better understandable? Thanks!

File: README.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+
+1) Contributors gather sentences from public domain sources and (optionally) pre-process and pre-review them. These sentences can be from public domain books, or even self-written. The source does not matter, as long as the sentences are in the public domain. Contributors then upload these sentences through the [Sentence Collector "Add" form](https://commonvoice.mozilla.org/sentence-collector/#/add)
+2) The Sentence Collector validates these sentences based on [rules per language](server/lib/validation/VALIDATION.md) (or the English rule file as default). Any sentence that does not match the validation rules does not get further processed and is shown as error in the Sentence Collector user interface for correction. For example: sentences are not allowed to have numbers in them, such as `2022`
+3) Any sentence that passed the validation gets written to the Sentence Collector database
```
Comment (CapitainFlam) [inline README.md:17]:
```suggestion
3) Any sentence that passed the validation gets written to the Sentence Collector database.
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+5) Contributor's reviews are saved in the Sentence Collector database. Sentences can be approved or rejected. If at least 2 out of 3 reviews are positive, the sentence will eventually be exported for Common Voice (see the steps below).
+6) Once a week an automatic process is triggered (GitHub action) to export all approved sentences to the Common Voice repository.
+7) During this export, the [cleanup](https://github.com/common-voice/sentence-collector/blob/main/server/lib/cleanup/CLEANUP.md) scripts are run for each sentence, if configured for a language. This can be used to apply transformations for consistency, such as converting "..." into "…".
+8) The resulting `sentence-collector.txt` file is written to the [language specific folder](https://github.com/common-voice/common-voice/tree/main/server/data) in the Common Voice repository. Note that any change to that file within the Common Voice repository will be overwritten by the next export, as the only source is the Sentence Collector database.
+9) Whenever a new version of the Common Voice website is released, the sentences get imported into the Common Voice database.
```
Comment (CapitainFlam) [inline README.md:23]: ...This sentence is unclear, I don't catch what it means... #NoClue 😸

File: README.md [hunk 3]
```diff
@@ -1,6 +1,27 @@
...
+4) These sentences then get shown on the [Sentence Collector "Review" page](https://commonvoice.mozilla.org/sentence-collector/#/review) for other contributors to review.
+5) Contributor's reviews are saved in the Sentence Collector database. Sentences can be approved or rejected. If at least 2 out of 3 reviews are positive, the sentence will eventually be exported for Common Voice (see the steps below).
+6) Once a week an automatic process is triggered (GitHub action) to export all approved sentences to the Common Voice repository.
+7) During this export, the [cleanup](https://github.com/common-voice/sentence-collector/blob/main/server/lib/cleanup/CLEANUP.md) scripts are run for each sentence, if configured for a language. This can be used to apply transformations for consistency, such as converting "..." into "…".
+8) The resulting `sentence-collector.txt` file is written to the [language specific folder](https://github.com/common-voice/common-voice/tree/main/server/data) in the Common Voice repository. Note that any change to that file within the Common Voice repository will be overwritten by the next export, as the only source is the Sentence Collector database.
```
Comment (CapitainFlam) [inline README.md:22]: It is 👍 to ⚠️ warn people about this !

**Review by CapitainFlam — COMMENTED** [review]
Looks ✨ 😻 🎇  ! 

I made (many ?) comment, but it's just clarification / desambiguation to get the message as clear as possible. Please have a look :)

File: README.md [hunk 4]
```diff
@@ -1,6 +1,27 @@
 # Common Voice Sentence Collector
 
-The [Sentence Collector](https://commonvoice.mozilla.org/sentence-collector/) is part of the [Common Voice](https://commonvoice.mozilla.org/) project. Its purpose is to provide a tool for contributors to upload public domain sentences, which then can get reviewed and are exported to the Common Voice database. Once imported they will show up for contributors on Common Voice to read out aloud.
+The [Sentence Collector](https://commonvoice.mozilla.org/sentence-collector/) is part of the [Common Voice](https://commonvoice.mozilla.org/) project. Its purpose is to provide a tool for contributors to upload public domain sentences, which then can get reviewed and are exported to the Common Voice database. Once imported into the Common Voice website, they will show up for contributors to read out aloud.
+
```
[the remark this answers is not in the document]
Comment (MichaelKohler) [reply README.md:4]: > ...By the way, we shall add (and adapt) the intro in Sentece Extractor's and Bulk Submission's repos too !

The Sentence Extractor has its own README here: https://github.com/Common-Voice/cv-sentence-extractor . And the Bulk Submission is documented here: https://github.com/common-voice/common-voice/blob/main/docs/SENTENCES.md#bulk-submission (there is no separate repository for bulk submissions)

File: README.md [hunk 5]
```diff
@@ -1,6 +1,27 @@
...
+![Diagram](docs/flow.svg)
+
+*To edit this diagram, load the `flow.svg` in the docs of the repository into [diagrams.net](https://app.diagrams.net/) and then save the updated version back into the repository like any other file changes you'd make.*
+
+In the diagram above, light blue squares represent Sentence Collector processes. The grey squares are processes outside of the Sentence Collector tooling. The grey processes are the same for other sentence sources, such as bulk submissions and Sentence Extractor. Instead of an automatic export, these use Pull Requests directly adding text files into the `server/data` folder of the Common Voice repository.
+
+1) Contributors gather sentences from public domain sources and (optionally) pre-process and pre-review them. These sentences can be from public domain books, or even self-written. The source does not matter, as long as the sentences are in the public domain. Contributors then upload these sentences through the [Sentence Collector "Add" form](https://commonvoice.mozilla.org/sentence-collector/#/add)
```
[the remark this answers is not in the document]
Comment (MichaelKohler) [reply README.md:15]: I don't think this is necessary. If you're not logged in, you automatically get redirected to the login page.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,27 @@
...
+
+## Detailed Flow
+
+This explanation only focuses on the Sentence Collector. For bulk uploads of thousands of sentences, Sentence Collector is not the best tool. Check out the [Bulk Submission](https://github.com/common-voice/common-voice/blob/main/docs/SENTENCES.md#bulk-submission) guidelines for this use case. Another tool is the [Sentence Extractor](https://github.com/Common-Voice/cv-sentence-extractor) which allows automatic extraction of data sources such as Wikipedia.
+
+![Diagram](docs/flow.svg)
+
+*To edit this diagram, load the `flow.svg` in the docs of the repository into [diagrams.net](https://app.diagrams.net/) and then save the updated version back into the repository like any other file changes you'd make.*
+
+In the diagram above, light blue squares represent Sentence Collector processes. The grey squares are processes outside of the Sentence Collector tooling. The grey processes are the same for other sentence sources, such as bulk submissions and Sentence Extractor. Instead of an automatic export, these use Pull Requests directly adding text files into the `server/data` folder of the Common Voice repository.
```
[the remark this answers is not in the document]
Comment (MichaelKohler) [reply README.md:13]: That's correct, there is no tooling around bulk submissions.

**Comment by MichaelKohler** [comment]
Thanks for the review! I have clarified a few points based on your feedback. And thanks for bringing this up, I think this makes the purpose and flow way easier to understand.

**Pull request merged**